workers/data_refinery_workers/processors/test_smasher.py

```diff
@@ -0,0 +1,348 @@
+            self.assertEqual(final_context['dataset'].is_processed, True)
+
+            dataset = Dataset.objects.filter(id__in=relations.values('dataset_id')).first()
+            dataset.is_processed = False
+            dataset.save()
+
+            # Cleanup
+            os.remove(final_context['output_file']) 
+
+        for scale_type in ['NONE', 'MINMAX', 'STANDARD', 'ROBUST']:
+            dataset = Dataset.objects.filter(id__in=relations.values('dataset_id')).first()
+            dataset.aggregate_by = 'EXPERIMENT'
+            dataset.scale_by = scale_type
+            dataset.save()
+
+            print ("Smashing " + scale_type)
+            final_context = smasher.smash(job.pk, upload=False)
+            # Make sure the file exists and is a valid size
+            self.assertNotEqual(os.path.getsize(final_context['output_file']), 0)
+            self.assertEqual(final_context['dataset'].is_processed, True)
+
+            dataset = Dataset.objects.filter(id__in=relations.values('dataset_id')).first()
+            dataset.is_processed = False
+            dataset.save()
+
+            # Cleanup
+            os.remove(final_context['output_file'])
+
+        # Stats
+        for scale_type in ['MINMAX', 'STANDARD']:
+            dataset = Dataset.objects.filter(id__in=relations.values('dataset_id')).first()
+            dataset.aggregate_by = 'EXPERIMENT'
+            dataset.scale_by = scale_type
+            dataset.save()
+
+            print("###")
+            print("# " + scale_type)
+            print('###')
+
+            final_context = smasher.smash(job.pk, upload=False)
+            final_frame = final_context['final_frame']
+            print("MEAN (per row):")
+            print(final_frame.mean(axis=1))
+            print("")
+
+            print("MIN (per row):")
+            print(final_frame.min(axis=1))
+            print("")
+
+            print("Max (per row):")
+            print(final_frame.max(axis=1))
+            print("")
+
+            print("STD (per row):")
+            print(final_frame.std(axis=1))
+            print("")
+
+            print("MEDIAN (per row):")
+            print(final_frame.median(axis=1))
+            print("")
+
+            os.remove(final_context['output_file'])
+
+
+    @tag("smasher")
+    def test_get_results(self):
+        """ Test our ability to collect the appropriate samples. """
+
+        sample = Sample()
+        sample.accession_code = 'GSM45588'
+        sample.save()
+
+        result = ComputationalResult()
+        result.save()
+
+        computed_file = ComputedFile()
+        computed_file.filename = "oh_boy.txt"
+        computed_file.result = result
+        computed_file.size_in_bytes = 123
+        computed_file.save()
+
+        assoc = SampleResultAssociation()
+        assoc.sample = sample
+        assoc.result = result
+        assoc.save()
+
+        computed_files = sample.get_result_files()
+        self.assertEqual(computed_files.count(), 1)
```

Comment:
This test seems like it might belong in common since it is strictly dealing with ORM classes and their methods.

Reply:
I like these here since it's related to the logic mentioned in the comment above.

Reply:
Word